README.md

```diff
@@ -4,18 +4,29 @@ A java based file attribute caching filesystem library
 This library is meant to wrap another filesystem in order to intercept and cache calls for reading and setting file
 attributes.
 
-It reduces OTHER_IOPS calls as listed in the "other" column when executing [Procmon] and looking at
+It reduces `OTHER_IOPS` calls as listed in the "other" column when executing [Procmon] and looking at
 operations performed on a given file where file attributes are read and/or set.
 
 ## Usage Instructions
-Dependency to add via gradle:
+Ensure your `settings.gradle.kts` file can pull dependencies from maven central.
+```kotlin
+dependencyResolutionManagement {
+    repositories {
+        mavenCentral()
+    }
+}
+```
+
+Dependency to add to your top level `build.gradle.kts` file:
```

Comment:
This is out of scope for documentation. I think supplying something like 

```
implementation("com.pkware.filesystem:file-attribute-caching:VERSION_HERE")
testImplementation("com.pkware.filesystem:file-attribute-caching:VERSION_HERE")
```

is fine for a quick copy/paste, but it's not our role to document Gradle, let alone such a basic function of Gradle.

Reply:
these sections im going to keep as this is open source, they are small yet helpful additions and I dont see these instructions needing to be updated anytime soon

Reply:
In light of this being a public / open source repo, I think concise documentation is better.
In this case, concise means specific to this code. For example, lines 11-19 are not.

Reply:
ehh im still against that, as standalone instructions for a standalone open source thing I would assume someone coming for this would want basic instructions of how to install it/use it

i have seen too many public repos that dont have any instructions for installation and then you have to go through maven/gradle docs- which are by no means simple to do for a new user

i dont think theres anything confusing about whats listed even though its not project specific this project will be published on maven rather than google or another public publisher

Reply:
It might even be worth something like a badge notation, such as those used on https://github.com/apache/logging-log4j2. Check out (https://shields.io/category/dependencies). This is more usable and provides guidance on _multiple_ languages rather than going into detail of only one.

Reply:
> ehh im still against that, as standalone instructions for a standalone open source thing I would assume someone coming for this would want basic instructions of how to install it/use it
> 
> i have seen too many public repos that dont have any instructions for installation and then you have to go through maven/gradle docs- which are by no means simple to do for a new user

But our documentation isn't geared towards teaching you how to import dependencies. Nobody seeking use of our library is at their first idiomatic rodeo.

> i dont think theres anything confusing about whats listed even though its not project specific this project will be published on maven rather than google or another public publisher

It's about brevity. We don't teach them where to download a JVM, or how to include Kotlin in their project. Why? Because it's out of scope.

Things that are more relevant for documentation would be educating someone on using this from _Java_, which is not covered.

Reply:
Since when do I have to apply the assumption about what the target audience is for an open source library? we dont have any standards for this as a company currently.

Ideally as a developer looking at a third party library I would want the following from a library readme at the very least:

- installation instructions for the library in the language it's written in along with the source its being published from
- usage instructions in the language it was written in
- release instructions for the repo it is stored on

anything else is a bonus- none of the other projects we have go into using different languages unless its an official sdk or the like where that is expected

Reply:
> Since when do I have to apply the assumption about what the target audience is for an open source library? we dont have any standards for this as a company currently.

Almost every time you write javadocs, I remind you that the target audience is an advanced (or at the very least intermediate) developer. 

If I'm evaluating a 3rd-party tool, I highly value good documentation and good tests. As @MatthewFoster624 alluded to, part of good documentation is providing only the necessary.

You're saying we don't have a standard, and I'm trying to establish one based on how the better open source projects tend to write their documentation, which educates users on _their_ offering. This isn't an exhaustive how-to implement with our code from start to finish, it's an opportunity to give a quick overview things developers typically look for.

I was on the fence about suggesting moving the `Release` instructions to `RELEASING.MD` for that very reason, but did not.

> anything else is a bonus- none of the other projects we have go into using different languages unless its an official sdk or the like where that is expected

Right, so are you saying that educating someone on how to use Java from Kotlin is out of scope? So is importing `mavenCentral` in Gradle. But educating on how to use this library from Java is FAR more helpful documentation than a trivial Gradle command.

I know you think you're providing a bonus, but you're mostly insulting the reader IMO.

Reply:
I should take the time to say that I think your initial documentation does a fantastic job of what I'm advocating on behalf of!